ConnectorAutocadCivil/ConnectorAutocadCivil/Utils.cs

```diff
@@ -254,95 +254,193 @@ public static bool Visible(this DBObject obj)
+          {
+            Color = Color.FromColorIndex(ColorMethod.ByColor, 7), // white
+            Name = cleanName
+          };
+
+        // Append the new layer to the layer table and the transaction
+        try
+        {
+          layerTable.Add(newLayer);
+        }
+        catch // TODO: use !IsFatal() here
+        {
+          return false;
+        }
+        tr.AddNewlyCreatedDBObject(newLayer, true);
+      }
+    }
+    return true;
+  }
+
+  #endregion
+
+  #region property sets
 #if CIVIL2021 || CIVIL2022 || CIVIL2023 || CIVIL2024
   private static Autodesk.Aec.PropertyData.DataType? GetPropertySetType(object prop)
   {
     switch (prop)
     {
-      case IEnumerable<string> _:
-      case IEnumerable<int> _:
-      case IEnumerable<double> _:
-      case IEnumerable<bool> _:
+      case IEnumerable<string>:
+      case IEnumerable<int>:
+      case IEnumerable<double>:
+      case IEnumerable<bool>:
         return Autodesk.Aec.PropertyData.DataType.List;
 
-      case string _:
+      case string:
         return Autodesk.Aec.PropertyData.DataType.Text;
-      case int _:
+      case int:
         return Autodesk.Aec.PropertyData.DataType.Integer;
-      case double _:
+      case double:
         return Autodesk.Aec.PropertyData.DataType.Real;
-      case bool _:
+      case bool:
         return Autodesk.Aec.PropertyData.DataType.TrueFalse;
 
       default:
         return null;
     }
   }
 
-  public static void SetPropertySets(this Entity entity, Document doc, List<Dictionary<string, object>> propertySetDicts)
+  public static PropertySetDefinition CreatePropertySet(Dictionary<string,object> propertySetDict, Document doc)
   {
-    // create a dictionary for property sets for this object
-    var name = $"Speckle {entity.Handle} Property Set";
-    int count = 0;
-    foreach (var propertySetDict in propertySetDicts)
-    {
-      // create the property set definition for this set.
-      var propSetDef = new PropertySetDefinition();
-      propSetDef.SetToStandard(doc.Database);
-      propSetDef.SubSetDatabaseDefaults(doc.Database);
-      var propSetDefName = name += $" - {count}";
-      propSetDef.Description = "Property Set Definition added with Speckle";
-      propSetDef.AppliesToAll = true;
-
-      // Create the definition for each property
-      foreach (var entry in propertySetDict)
+    PropertySetDefinition propSetDef = new();
+    propSetDef.SetToStandard(doc.Database);
+    propSetDef.SubSetDatabaseDefaults(doc.Database);
+    propSetDef.Description = "Property Set Definition added with Speckle";
+    propSetDef.AppliesToAll = true;
+
+    // Create the definition for each property
+    foreach (var entry in propertySetDict)
+    {
+      var propDef = new PropertyDefinition();
+      propDef.SetToStandard(doc.Database);
+      propDef.SubSetDatabaseDefaults(doc.Database);
+      propDef.Name = entry.Key;
+      if (GetPropertySetType(entry.Value) is Autodesk.Aec.PropertyData.DataType dataType)
```

Comment:
What if the type doesn't maych, we add a propDef with a null? DataType?

Reply:
`DataType` is an enum, so I believe it will default to `int`. This is a good catch though, I will skip this if the type cannot be determined and log it instead.